File: Ridy-taxi-app/android/driver/src/main/java/com/innomalist/taxi/driver/networking/socket/GetStats.kt
package com.innomalist.taxi.driver.networking.socket

import com.innomalist.taxi.common.networking.socket.interfaces.SocketRequest
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

class GetStats(queryType: QueryType) : SocketRequest() {
    init {
        this.params = arrayOf(queryType.value)
    }
}

@JsonClass(generateAdapter = true)
data class StatisticsResult(
        var currency: String,
        var dataset: List<DataPoint>
)

@JsonClass(generateAdapter = true)
data class DataPoint(
        var name: String,
        var current: String,
        var earning: Float,
        var count: String,
        var distance: String,
        var time: String
)

enum class QueryType(val value: String) {
    @Json(name = "daily")
    Daily("daily"),
    @Json(name = "weekly")
    Weekly("weekly"),
    @Json(name = "monthly")
    Monthly("monthly");

    companion object {
        operator fun get(code: String): QueryType {
            for (s in values()) {
                if (s.value == code) return s
            }
            return Daily
        }
    }
}

File: Ridy-taxi-app/ios/driver/driver/Networking/WebSocket/GetStats.swift
//
//  GetStats.swift
//  driver
//
//  Created by Manly Man on 1/20/20.
//  Copyright © 2020 minimal. All rights reserved.
//

import UIKit

class GetStats: SocketRequest {
    typealias ResponseType = StatisticsResult
    public var params: [Any]?
    
    init(query: QueryType) {
        self.params = [query.rawValue]
    }
}

struct StatisticsResult: Codable {
    var currency: String
    var dataset: [DataPoint]
}

struct DataPoint: Codable {
    var name: String
    var current: String
    var earning: Double
    var count: String
    var distance: String
    var time: String
}

enum QueryType: String, Codable {
    case Daily = "daily"
    case Weekly = "weekly"
    case Monthly = "monthly"
}
